package.json

```diff
@@ -2,19 +2,47 @@
   "name": "nuxt-app",
   "private": true,
   "type": "module",
+  "engines": {
+    "node": "20.15.0"
+  },
+  "lint-staged": {
+    "./**/*.{js,jsx,ts,tsx,vue}": [
+      "npx eslint . --cache --fix",
+      "npx prettier --write"
+    ],
+    "./**/*.{json,css}": [
+      "npx prettier --write"
+    ]
+  },
   "scripts": {
     "build": "nuxt build",
     "dev": "nuxt dev --port 8080",
-    "generate": "nuxt generate",
     "preview": "nuxt preview --port 8080",
-    "postinstall": "nuxt prepare"
+    "postinstall": "nuxt prepare",
+    "format": "prettier \"**/*.{js,ts,jsx,tsx,vue}\" --write",
+    "generate": "nuxt generate",
+    "link-module": "pnpm link ../gc-shared-components",
+    "lint:prettier": "prettier ./ --check --ignore-path .prettierignore",
+    "lint:eslint": "eslint . --cache",
+    "lint:eslint:fix": "eslint . --cache --fix",
+    "lint": "pnpm run lint:prettier && pnpm run lint:eslint",
+    "lint:fix": "pnpm run lint:eslint:fix && pnpm run format"
   },
   "dependencies": {
     "@auth0/auth0-spa-js": "^2.2.0",
     "@tailwindcss/vite": "^4.1.10",
+    "eslint": "^9.31.0",
+    "eslint-plugin-vue": "^10.3.0",
     "nuxt": "^3.17.5",
     "tailwindcss": "^4.1.10",
     "vue": "^3.5.16",
     "vue-router": "^4.5.1"
+  },
+  "packageManager": "pnpm@10.12.2+sha512.a32540185b964ee30bb4e979e405adc6af59226b438ee4cc19f9e8773667a66d302f5bfee60a39d3cac69e35e4b96e708a71dd002b7e9359c4112a1722ac323f",
+  "devDependencies": {
+    "@eslint/js": "^9.31.0",
+    "eslint-config-flat-gitignore": "^2.1.0",
+    "globals": "^16.3.0",
+    "typescript-eslint": "^8.36.0"
```

Comment:
This seems different than the approach in `gc-explorer` to use `typescript` directly.

I have to admit that trying to figure out typescript and eslint with Nuxt has given me headaches before. It's not at all obvious what is the right approach.

If this approach you've taken here works to lint TS in the repo, then we can adapt the same for `gc-explorer` and the other Nuxt repos.

Reply:
Can you open a quick PR in `gc-explorer` with the same changes? That'd be much appreciated!